### PR TITLE
Add script to validate urls in .md

### PR DESCRIPTION
**Pull request by zhenghuiwang**
Related to https://github.com/kubeflow/website/issues/1683

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1684)

### CODE REVIEW

File: scripts/validate-urls.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+parser.add_argument(
+    '-d',
+    '--dir',
+    dest='input_dir',
+    nargs='?',
+    default='kubeflow/website/content',
+    help=
+    'Path to the doc conetnt folder. (default: %(default)s)',
```
Comment (sarahmaddox) [inline scripts/validate-urls.py]: Typo: "conetnt" > "content"

Also, I think "default" should start with a capital letter: 
`(Default: %(default)s)`

File: scripts/validate-urls.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# This script extract http/https URLs from .md files and validate them.
```
Comment (sarahmaddox) [inline scripts/validate-urls.py]: Can you add some usage instructions at the top, just to get people started. For example, where should I run this script and what are the required parameters (if any). An example of the basic usage would be great.

Comment (sarahmaddox) [inline scripts/validate-urls.py]: s/extract/extracts
s/validate/validates

File: scripts/validate-urls.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+def should_skip(url):
+    for p in WHITE_LIST:
+        if p.match(url):
+            return True
+    return False
+
+def main():
+    args = parser.parse_args()
+    # find all md file under INPUT_DIR.
```
Comment (sarahmaddox) [inline scripts/validate-urls.py]: s/file/files

File: scripts/validate-urls.py (hunk 2 above)
Comment (zhenghuiwang) [reply scripts/validate-urls.py]: I added more comments. PTAL. Thanks!

File: scripts/validate-urls.py [hunk 4]
```diff
@@ -1,4 +1,22 @@
-# This script extract http/https URLs from .md files and validate them.
+# This script finds .md files under a directory and its subdirectories, extracts
+# http/https URLs from .md files and validates them.
+#
+# This script can be run peoriodically on kubeflow/website source repository
```
Comment (sarahmaddox) [inline scripts/validate-urls.py]: s/peoriodically/periodically

File: scripts/validate-urls.py [hunk 5]
```diff
@@ -1,4 +1,22 @@
...
+# to find outdated URLs, which indicate possible outdated document sections.
+#
+# To run this script, type
+#   $ python3.8 validate-urls.py -d /path/to/kubeflow/website/content/docs
```
Comment (sarahmaddox) [inline scripts/validate-urls.py]: I think it'd be better to remove the $ sign. People may think they need to include it. Maybe say something like this:

```
# To run this script, type the following on the command line:
# python3.8 validate-urls.py -d /path/to/kubeflow/website/content/docs
```

**Comment by sarahmaddox** [comment]
Related issue: https://github.com/kubeflow/website/issues/73

Thanks @zhenghuiwang!
/lgtm
/approve

**Pull request merged**